ouroboros-consensus/src/Ouroboros/Storage/ChainDB/Model.hs

```diff
@@ -58,11 +56,12 @@ import           Ouroboros.Storage.ChainDB.API (IteratorId (..),
 
 -- | Model of the chain DB
 data Model blk = Model {
-      blocks        :: Map (HeaderHash blk) blk
-    , cps           :: CPS.ChainProducerState blk
-    , currentLedger :: ExtLedgerState blk
-    , initLedger    :: ExtLedgerState blk
-    , iterators     :: Map IteratorId [blk]
+      blocks          :: Map (HeaderHash blk) blk
+    , cps             :: CPS.ChainProducerState blk
+    , currentLedger   :: ExtLedgerState blk
+    , initLedger      :: ExtLedgerState blk
+    , iterators       :: Map IteratorId [blk]
+    , validationCache :: ValidationCache blk
```

Comment:
It seems that there should be no need for this cache; instead the computation of `chains` should be fixed not to construct prefixes of valid chains (justified by the "always extend" property).